iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/relative_pointer.hpp

```diff
@@ -64,23 +64,27 @@ class RelativePointer : public BaseRelativePointer
     template <typename U = T>
     typename std::enable_if<!std::is_void<U>::value, U&>::type operator*() noexcept;
 
-    /// @brief access to the underlying object
-    /// @return a pointer to the underlying object
-    T* operator->() noexcept;
-
     /// @brief dereferencing operator which returns a const reference to the underlying object
     /// @tparam U a template parameter to enable the dereferencing operator only for non-void T
     /// @return a const reference to the underlying object
     template <typename U = T>
-    typename std::enable_if<!std::is_void<U>::value, const U&>::type operator*() const noexcept;
+    const typename std::enable_if<!std::is_void<U>::value, const U&>::type operator*() const noexcept;
+
+    /// @brief access to the underlying object
+    /// @return a pointer to the underlying object
+    T* operator->() noexcept;
 
     /// @brief read-only access to the underlying object
     /// @return a const pointer to the underlying object
-    T* operator->() const noexcept;
+    const T* operator->() const noexcept;
 
     /// @brief access the underlying object
     /// @return a pointer to the underlying object
-    T* get() const noexcept;
+    T* get() noexcept;
```

Comment:
@MatthiasKillat So you mean: Only 

```cpp
    T* get() const noexcept;
```

instead of

```cpp
    T* get() noexcept;

    const T* get() const noexcept;
```

?

Reply:
That is how `shared_ptr` does it, unless I miss something. And I can see why (as said before). Not super important but we have to keep in mind that it can be instantiated with `const T` and `T` for any `T` which is logically responsible for constness  of the return type.

Reply:
Reverted the changes.